### PR TITLE
Add parser

### PR DESCRIPTION
**Pull request by lebacql**
Simple array parser
String json parser

### CODE REVIEW

File: src/Parser/JsonPaser.php [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+ * Date: 06/11/18
+ * Time: 11:03
+ */
+
+namespace MetaHydrator\Parser;
+
+
+use MetaHydrator\Exception\ParsingException;
+
+class JsonPaser extends AbstractParser
```
Comment (moufmouf) [inline src/Parser/JsonPaser.php]: This class is not needed

File: src/Parser/SimpleArrayParser.php [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+class SimpleArrayParser extends AbstractParser
+{
+    /**
+     * @param $rawValue
+     * @return mixed|null
+     * @throws \MetaHydrator\Exception\ParsingException
+     */
+    public function parse($rawValue)
+    {
+        if ($rawValue === null || $rawValue === '') {
```
Comment (moufmouf) [inline src/Parser/SimpleArrayParser.php]:
```suggestion
        if ($rawValue === null) {
```

There is no need to check against empty string, this should be invalid (i.e it should throw an exception)

File: src/Parser/SimpleArrayParser.php [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+            return null;
+        } else {
```
Comment (moufmouf) [inline src/Parser/SimpleArrayParser.php]: You can remove the else statement entirely (this limits indentation)

File: src/Parser/SimpleArrayParser.php [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+            $value = is_array($rawValue);
```
Comment (moufmouf) [inline src/Parser/SimpleArrayParser.php]: Rename `$value` in something else (for instance: `$isArray`)

File: tests/Parser/SimpleArrayParserTest.php [hunk 5]
```diff
@@ -0,0 +1,28 @@
+<?php
+/**
```
Comment (moufmouf) [inline tests/Parser/SimpleArrayParserTest.php]: Can you remove this header?

File: tests/Parser/SimpleArrayParserTest.php [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+    {
+        $parser = new SimpleArrayParser();
+
+        try {
+            $parser->parse('');
+            $parser->parse(["random", "lambda", 42]);
+            $parser->parse(null);
+        }catch (\Exception $e){
+            echo $e;
+        }
```
Comment (moufmouf) [inline tests/Parser/SimpleArrayParserTest.php]: Write 3 tests for the 3 use cases.
Use `$this->assertSame(..., ...)` (for null, and array)
Use `$this->expectException(....)` (to expect a SPECIFIC exception)

File: tests/Parser/SimpleArrayParserTest.php [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+    public function testParseSimpleArray()
+    {
+        $parser = new SimpleArrayParser();
+
+        $this->assertSame(["random", "lambda", 42, null], $parser->parse(["random", "lambda", 42, null]));
+        $this->assertSame(["random", "lambda", 42], $parser->parse(["random", "lambda", 42]));
+        $this->assertSame(["random", "lambda"], $parser->parse(["random", "lambda"]));
+        $this->assertSame([21, 42], $parser->parse([21, 42]));
+        $this->assertSame([42], $parser->parse([42]));
+        $this->assertSame([], $parser->parse([]));
```
Comment (moufmouf) [inline tests/Parser/SimpleArrayParserTest.php]: Only one line is needed

File: tests/Parser/SimpleArrayParserTest.php [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+
+
+        $this->expectException(ParsingException::class);
+        $parser->parse('');
+
+        $this->expectException(ParsingException::class);
+        $parser->parse(null);
```
Comment (moufmouf) [inline tests/Parser/SimpleArrayParserTest.php]: Will not return null.

File: tests/Parser/SimpleArrayParserTest.php (hunk 7 above)
Comment (moufmouf) [reply tests/Parser/SimpleArrayParserTest.php]: You need to add a test on `null
